### DISCUSSION TITLE
Made simple framework to write widgets in lua

### DISCUSSION THREAD

**Discussion by Horryportier** [elkowar/eww#976] [Show and tell]
HI i've made simple framework that wraps around eww and can create widgets with lua. 
https://github.com/Horryportier/lua-eww

It works by using `literal` widget and `defpoll` variables to request widgets from lua script. 
Generating widgets is as simple as writing table with that are available to widget type and parsing it into something yuck can understand.
As an demo i've made simple widget that list your home dir and allows you to open files and directories everything in lua. 

<img src="https://raw.githubusercontent.com/Horryportier/lua-eww/main/demo/home_dir.png" width=10% height=10%>

I've implemented  it in a way to be used however you want u can have some written in your standard eww.yuck , you can put multiple  widgets into one window or into separate or, you can just use it to generate widget print them to the terminal and put in your config. DO AS YOU WISH!    

If there are is something you want to be implement  or you find an bug please make issue 

Hope you like it

**Comment by ItsTerm1n4l** [comment]
Interesting...
